framework/src/base/DependencyResolverInterface.C

```diff
@@ -0,0 +1,27 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "DependencyResolverInterface.h"
+
+template<>
+InputParameters
+validParams<DependencyResolverInterface>()
+{
+  InputParameters params = emptyInputParameters();
+  return params;
+}
+
+DependencyResolverInterface::DependencyResolverInterface(const InputParameters & parameters)
```

Comment:
My original plan was to add parameters to this class so that the iterative warning and debug flags (see FEProblem and DebugOutputAction) could reside here. Well that is not possible. But, I decided to keep the addition of InputParameters to the interface class for consistency with other classes and to allow future parameter settings. If someone is against this I can pull it out.